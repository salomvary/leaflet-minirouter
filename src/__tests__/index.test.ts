import * as L from "leaflet";
import Layer from "../layer";
import {
  fireEvent,
  getByAltText,
  waitForElement,
  wait
} from "@testing-library/dom";
import * as mapboxResponse from "./mapbox-response.json";

test("creating a new route", async () => {
  // SVGSVGElement.prototype.createSVGRect = () => <DOMRect>null;
  function svgCreate(name) {
    return document.createElementNS("http://www.w3.org/2000/svg", name);
  }
  var svg = !!(document.createElementNS && svgCreate("svg").createSVGRect);
  // const svgEl = svgCreate("svg");

  const lsvg = L.Browser.svg;
  window.fetch = jest.fn();
  (<jest.Mock>window.fetch).mockResolvedValueOnce({
    json: () => Promise.resolve(mapboxResponse)
  });
  const container = document.createElement("div");
  container.style.width = "1000px";
  container.style.height = "1000px";
  const map = L.map(container).setView([50, 0], 13);
  const layer = new Layer();
  layer.addTo(map);
  fireEvent.click(container, { clientX: 100, clientY: 100 });
  fireEvent.click(container, { clientX: 200, clientY: 200 });

  expect(getByAltText(container, "Route start")).toBeTruthy();
  expect(getByAltText(container, "Route end")).toBeTruthy();

  // Wait until the route path is rendered into an svg layer
  await wait(() => {
    expect(container.querySelector("svg")).toBeTruthy();
  });

  // XXX the <path m> in the snapshot seems wrong, maybe
  // the overall map dimensions are not detected properly
  expect(container.querySelector("svg")).toMatchSnapshot();
});
