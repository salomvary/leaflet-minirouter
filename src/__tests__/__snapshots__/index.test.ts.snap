// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating a new route 1`] = `
<svg
  height="0"
  pointer-events="none"
  style="left: 0px; top: 0px;"
  viewBox="0 0 0 0"
  width="0"
>
  <g>
    <path
      class="leaflet-interactive"
      d="M0 0"
      fill="none"
      stroke="#3388ff"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-opacity="1"
      stroke-width="3"
    />
  </g>
</svg>
`;
